File: bankingtool.client/src/app/dto/request/bank-account/create-account-request.ts
import { RequestId } from "../request-id";

export interface CreateAccountRequestObject {
  request: CreateAccountRequest;
}

export interface CreateAccountRequest {
  bank: CreateAccountRequestBank;
  account: CreateAccountRequestAccount;
  customer: CreateAccountRequestCustomer;
}

export interface CreateAccountRequestBank extends RequestId {

}

export interface CreateAccountRequestCustomer extends RequestId {
  customerWantCreditCard: boolean;
}

export interface CreateAccountRequestAccount {
  accountTypeId: number;
  doYouWantToChangeThisAccountToPrimaryAccount: boolean;
}

File: BankingTool.Model/Dto/RequestDto/CreateAccountRequest.cs
namespace BankingTool.Model.Dto.RequestDtos
{
    public class CreateAccountRequestObject
    {
        public CreateAccountRequest Request { get; set; }
    }
    public class CreateAccountRequest
    {
        public CreateAccountRequestBank Bank { get; set; }
        public CreateAccountRequestAccount Account { get; set; }
        public CreateAccountRequestCustomer Customer { get; set; }
    }
    public class CreateAccountRequestBank : RequestId
    {
    }
    public class CreateAccountRequestCustomer : RequestId
    {
        public bool CustomerWantCreditCard { get; set; }
    }
    public class CreateAccountRequestAccount
    {
        public int AccountTypeId { get; set; }
        public bool DoYouWantToChangeThisAccountToPrimaryAccount { get; set; }
    }
}
